cobaya/theory.py

```diff
@@ -204,11 +213,13 @@ def check_cache_and_compute(self, dependency_params=None, want_derived=False,
         (retrieved using get_current_derived()).
         """
         self.log.debug("Got parameters %r", params_values_dict)
-
-        for set_param in getattr(self, _requires, []):
+        params_values_dict = params_values_dict.copy()
+        for set_param in self.get_requirements():
```

Comment:
Perhaps we could precompute a list of which things in get_requirements are not in input_params and look like parameters, so the copy and loop can still be avoided in most cases?

Reply:
Done in 39bcf36 as a property.